lib/ecto_ldap/adapter.ex

```diff
@@ -44,6 +48,29 @@ defmodule Ecto.Ldap.Adapter do
     {:reply, search_response, state}
   end
 
+  def handle_call({:update, schema_meta, fields, filters}, _from, state) do
```

Comment:
All this needs to be moved a level up. Anything that gets sent to through the GenServer API needs to ready to go already.